src/nvim/plines.c

```diff
@@ -893,6 +897,11 @@ int plines_win_full(win_T *wp, linenr_T lnum, linenr_T *const nextp, bool *const
   if (foldedp != NULL) {
     *foldedp = folded;
   }
+
+  if (decor_conceal_line(wp, lnum - 1, false)) {
+    return 0;
+  }
+
```

Comment:
I think this early return is wrong. When a line is concealed, there may still be virtual lines attached to the line above (with `virt_lines_above = false`).

Reply:
Hmm the early return maybe should be wrong but I don't think it currently is. I think because the line is concealed `virt_lines_above = false` for the above line will not be drawn currently?

Reply:
Oh, right. Add a TODO?

Reply:
Fixed in https://github.com/neovim/neovim/compare/e67e399ea01b3198070a821423e73eab6f89133a..d1b63fa0ed482479bd7add4cf2a94dd2d88d0145.